src/tests/unit/mock-helpers/chrome-alarms-api-mock.ts

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// This must be the first import, otherwise importing webextension-polyfill from within
+// adapter implementations will fail
+import 'tests/unit/common/webextension-polyfill-setup';
+import { IMock, It, Mock, Times } from 'typemoq';
+import { Alarms, Events } from 'webextension-polyfill';
+
+export class ChromeAlarmsAPMIMock {
+    private underlyingMock: IMock<Alarms.Static> = Mock.ofType<Alarms.Static>();
+    private alarms: Alarms.Alarm[] = [];
+    private alarmListener: (name: Alarms.Alarm) => void;
+
+    public setupAddListener(callback: (name: Alarms.Alarm) => void, times = 1) {
+        const mockOnAlarm: IMock<Events.Event<(name: Alarms.Alarm) => void>> =
+            Mock.ofType<Events.Event<(name: Alarms.Alarm) => void>>();
+        Mock.ofType<(callback: (name: Alarms.Alarm) => void) => void>();
```

Comment:
This line seems to have no effect (based on running the tests with it commented out)- is it necessary?

Reply:
nice catch!